src/lxc/conf.c

```diff
@@ -1433,6 +1434,40 @@ static int lxc_setup_devpts(int num_pts)
 		DEBUG("removed existing \"/dev/ptmx\"");
 	}
 
+	/* Read attributes of /dev/pts/ptmx */
+	ret = stat("/dev/pts/ptmx", &ptmx);
+	if (ret) {
+		SYSERROR("failed to stat \"/dev/pts/ptmx\"");
+		return -1;
+	} else if (!S_ISCHR(ptmx.st_mode)) {
+		SYSERROR("\"/dev/pts/ptmx\" isn't a character device!");
+		return -1;
+	}
+
+	/* Create doppelganger of /dev/pts/ptmx at /dev/ptmx */
+	ret = mknod("/dev/ptmx", ptmx.st_mode, ptmx.st_rdev);
+	if (!ret) {
```

Comment:
The handler records - on startup - whether you are root or not. So you should either give this fun an additional arg to inform it whether it is root or not or pass down the handler itself and. Then you can check here and only perform the `mknod()` syscall when you are root. This saves a context switch.